library/src/lib/calendar/calendar-views/calendar2-year-view/calendar2-year-view.component.ts

```diff
@@ -1,17 +1,127 @@
-import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import { Component, OnInit, ViewEncapsulation, Output, Input, EventEmitter, ElementRef, AfterViewChecked } from '@angular/core';
 
 @Component({
     selector: 'fd-calendar2-year-view',
     templateUrl: './calendar2-year-view.component.html',
     styleUrls: ['./calendar2-year-view.component.scss'],
     encapsulation: ViewEncapsulation.None
 })
-export class Calendar2YearViewComponent implements OnInit {
+export class Calendar2YearViewComponent implements AfterViewChecked {
 
-    constructor() {
+    /** @hidden */
+    newFocusedYearId: string;
+
+    /** Parameter used for key */
+    @Input()
+    id: string;
+
+    /** Function that allows to specify which function would be called, when focus wants to escape */
+    @Input()
+    focusEscapeFunction: Function;
+
+    /** Parameter that stores the dozen of years that are currently being displayed */
+    @Input()
+    calendarYearList: number[];
+
+    /** Parameter holding the year that is currently selected */
+    @Input()
+    yearSelected: number;
+
+    /** Event fired when a year is selected */
+    @Output()
+    yearClicked: EventEmitter<any> = new EventEmitter();
+
+    /** Event fired when the next/previous dozen of years must be displayed */
+    @Output()
+    reGenerateYearList: EventEmitter<any> = new EventEmitter();
+
+    /** Parameter storing the year of the present day */
+    currentYear: number = new Date().getFullYear();
```

Comment:
Avoid using Date, ideally I want the only use of the Date object to be **inside** the FdDate object. Instead you can use FdDate.getToday().year IIRC